src/ApparentHorizons/StrahlkorperCoordsInDifferentFrame.cpp

```diff
@@ -55,38 +55,72 @@ void strahlkorper_coords_in_different_frame(
+      const auto& logical_to_grid_map = block.moving_mesh_logical_to_grid_map();
+      // Fill only those dest_cartesian_coords that are in this Block.
+      // Determine which coords are in this Block by checking if the
+      // inverse grid-to-logical map yields logical coords between -1 and 1.
+      for (size_t s = 0; s < get<0>(src_cartesian_coords).size(); ++s) {
+        const tnsr::I<double, 3, SrcFrame> x_src{
+            {get<0>(src_cartesian_coords)[s], get<1>(src_cartesian_coords)[s],
+             get<2>(src_cartesian_coords)[s]}};
+        const auto x_logical = logical_to_grid_map.inverse(x_src);
+        // x_logical might be an empty std::optional.
+        if (x_logical.has_value() and get<0>(x_logical.value()) <= 1.0 and
+            get<0>(x_logical.value()) >= -1.0 and
+            get<1>(x_logical.value()) <= 1.0 and
+            get<1>(x_logical.value()) >= -1.0 and
+            get<2>(x_logical.value()) <= 1.0 and
+            get<2>(x_logical.value()) >= -1.0) {
+          const auto x_dest =
+              grid_to_inertial_map(x_src, time, functions_of_time);
+          get<0>(*dest_cartesian_coords)[s] = get<0>(x_dest);
+          get<1>(*dest_cartesian_coords)[s] = get<1>(x_dest);
+          get<2>(*dest_cartesian_coords)[s] = get<2>(x_dest);
+          // Note that if a point is on the boundary of two or more
+          // Blocks, it might get filled twice, but that's ok.
+        }
+      }
+    } else {
+      static_assert(std::is_same_v<SrcFrame, ::Frame::Distorted>,
+                    "Src frame must be Distorted if it is not Grid");
+      const auto& grid_to_distorted_map =
+          block.moving_mesh_grid_to_distorted_map();
+      const auto& distorted_to_inertial_map =
+          block.moving_mesh_distorted_to_inertial_map();
+      const auto& logical_to_grid_map = block.moving_mesh_logical_to_grid_map();
+      // Fill only those dest_cartesian_coords that are in this Block.
+      // Determine which coords are in this Block by checking if the
+      // inverse distorted-to-logical map yields logical coords between -1
+      // and 1.
+      for (size_t s = 0; s < get<0>(src_cartesian_coords).size(); ++s) {
+        const tnsr::I<double, 3, SrcFrame> x_src{
+            {get<0>(src_cartesian_coords)[s], get<1>(src_cartesian_coords)[s],
+             get<2>(src_cartesian_coords)[s]}};
+        const auto x_grid =
+            grid_to_distorted_map.inverse(x_src, time, functions_of_time);
+        if (x_grid.has_value()) {
+          const auto x_logical = logical_to_grid_map.inverse(x_grid.value());
+          // x_logical might be an empty std::optional.
+          if (x_logical.has_value() and get<0>(x_logical.value()) <= 1.0 and
+              get<0>(x_logical.value()) >= -1.0 and
+              get<1>(x_logical.value()) <= 1.0 and
+              get<1>(x_logical.value()) >= -1.0 and
+              get<2>(x_logical.value()) <= 1.0 and
+              get<2>(x_logical.value()) >= -1.0) {
```

Comment:
This looks like it could suffer from the same roundoff error bug we had in BlockLogicalCoordinates (#4510). Perhaps apply the same fix that we have in `BlockLogicalCoordinates.cpp`? Same in the code branch above.